File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/repo/base/MarkerListManager.java
package com.example.ama.android2_lesson03.repo.base;

import com.example.ama.android2_lesson03.repo.data.model.SimpleMarker;

import java.util.ArrayList;

/**
 * Interface describes basic methods for managing markers by presenter's queries
 */
public interface MarkerListManager {
    void getAllMarkers(OnGetMarkersCallback callback);

    void updateMarker(SimpleMarker marker, String newName, OnGetMarkersCallback callback);

    void deleteMarker(SimpleMarker marker, OnGetMarkersCallback callback);

    void saveCurrentMarker(SimpleMarker marker);

    void prepareEditMarkerNameDialog(SimpleMarker marker, OnDialogPreparedCallback callback);

    interface OnGetMarkersCallback {
        void onSuccess(ArrayList<SimpleMarker> markers);
    }

    interface OnDialogPreparedCallback {
        void onSuccess(String dialogTitle, String dialogMessage, SimpleMarker targetMarker);
    }
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/repo/base/MarkerListManager.kt
package com.example.ama.android2_lesson03.repo.base

import com.example.ama.android2_lesson03.repo.model.SimpleMarker

/**
 * Interface describes basic methods for managing markers by presenter's queries
 */
interface MarkerListManager {
    fun getAllMarkers(callback: (markers: ArrayList<SimpleMarker>) -> Unit)
    fun updateMarker(marker: SimpleMarker, newName: String, callback: (markers: ArrayList<SimpleMarker>) -> Unit)
    fun deleteMarker(marker: SimpleMarker, callback: (markers: ArrayList<SimpleMarker>) -> Unit)
    fun saveCurrentMarker(marker: SimpleMarker)
    fun prepareEditMarkerNameDialog(marker: SimpleMarker, callback: (dialogName: String, dialogMessage: String, marker: SimpleMarker)-> Unit)
}